src/main/java/com/squareup/kotlinpoet/CodeBlock.kt

```diff
@@ -379,3 +379,8 @@ class CodeBlock private constructor(
         = c == '%' || c == '>' || c == '<' || c == '[' || c == ']' || c == 'W'
   }
 }
+
+@JvmOverloads
+fun Iterable<CodeBlock>.joinToCode(separator: CharSequence = ", ", prefix: CharSequence = "",
+                                   suffix: CharSequence = "" ) =
+    CodeBlock.of(map { it.toString() }.joinToString(separator, prefix, suffix))
```

Comment:
Calling `toString` prevents the ability to import types in these blocks.

It should be something like
```kotlin
val blocks = toTypedArray<CodeBlock>()
return CodeBlock.of(repeat("%L", blocks.size).joinToString(separator, prefix, suffix), *blocks)
```